app/platform/__init__.py

```diff
@@ -373,3 +382,156 @@ def edit_form(grant_id: UUID4, collection_id: UUID4, section_id: UUID4, form_id:
         db_form=db_form,
         form=wt_form,
     )
+
+
+@platform_blueprint.route(
+    "/grants/<uuid:grant_id>/developers/collections/<uuid:collection_id>/sections/<uuid:section_id>/forms/<uuid:form_id>/questions/add/choose-type",
+    methods=["GET", "POST"],
+)
+@mhclg_login_required
+def choose_question_type(
+    grant_id: UUID4, collection_id: UUID4, section_id: UUID4, form_id: UUID4
+) -> ResponseReturnValue:
+    db_form = get_form_by_id(form_id)
+    wt_form = QuestionTypeForm(question_data_type=request.args.get("question_data_type", None))
+    if wt_form.validate_on_submit():
+        question_data_type = wt_form.question_data_type.data
+        return redirect(
+            url_for(
+                "platform.add_question",
+                grant_id=grant_id,
+                collection_id=collection_id,
+                section_id=section_id,
+                form_id=form_id,
+                question_data_type=question_data_type,
+            )
+        )
+    return render_template(
+        "platform/developers/choose_question_type.html",
+        grant=db_form.section.collection_schema.grant,
+        collection=db_form.section.collection_schema,
+        section=db_form.section,
+        db_form=db_form,
+        form=wt_form,
+    )
+
+
+@platform_blueprint.route(
+    "/grants/<uuid:grant_id>/developers/collections/<uuid:collection_id>/sections/<uuid:section_id>/forms/<uuid:form_id>/questions/add",
+    methods=["GET", "POST"],
+)
+@mhclg_login_required
+@auto_commit_after_request
+def add_question(grant_id: UUID4, collection_id: UUID4, section_id: UUID4, form_id: UUID4) -> ResponseReturnValue:
+    form = get_form_by_id(form_id)
+    question_data_type_arg = request.args.get("question_data_type", None)
+    question_data_type_enum = QuestionDataType.coerce(question_data_type_arg)
```

Comment:
If the `question_data_type` is invalid I think this'll throw a 500, which ideally we wouldn't make super easy based on user input.

Probably a fallback value of text is easier, but the alternative is to render a 400 of some kind.

```suggestion
    question_data_type_arg = request.args.get("question_data_type", QuestionDataType.TEXT.name)
    question_data_type_enum = QuestionDataType.coerce(question_data_type_arg)
```